packages/tools/pie-icons-webc/build.js

```diff
@@ -1,88 +1,105 @@
-import pieIcons from '@justeattakeaway/pie-icons';
-import { pascalCase } from 'pascal-case';
+import pascalCase from 'just-pascal-case';
+import kebabCase from 'just-kebab-case';
 import path from 'path';
-import fs from 'fs-extra';
-import kebabCase from 'kebab-case';
+import fs from 'fs';
 
-import { normalizeIconName } from '@justeattakeaway/pie-icons-configs';
+import pieIcons from '@justeattakeaway/pie-icons';
+import {
+    normalizeIconName,
+    largeIconSizeDefault,
+    regularIconSizeDefault,
+} from '@justeattakeaway/pie-icons-configs';
 
 const { icons } = pieIcons.default;
 
 const componentTemplate = (name, svg) => {
-    const [, svgClasses] = svg.match(/class="(.+?)"/);
-    // The following styles make sure that if the icon is used inside pie-icon-button, it will be sized correctly
-    const styleTag = '<style>:host-context(pie-icon-button) svg, :host-context(pie-button) svg { display:block; width: var(--btn-icon-size); height: var(--btn-icon-size); }</style>';
-    return `import { getSvgProps } from '@justeattakeaway/pie-icons-configs';
-
-const template = document.createElement('template');
-template.innerHTML = '${styleTag}${svg}';
-
-export class ${name} extends HTMLElement {
-    constructor () {
-        super();
-        const clone = template.content.cloneNode(true);
-        this.root = this.attachShadow({ mode: 'open' });
-        this.root.append(clone);
-    }
+    const svgClasses = svg.match(/class="(.+?)"/)?.[1];
 
-    static get observedAttributes () {
-        return ['size'];
-    }
+    const isLargeIcon = name.endsWith('Large');
+    const sizeType = isLargeIcon ? 'LargeIconSize' : 'RegularIconSize';
 
-    get size () {
-        return this.getAttribute('size');
-    }
+    return `import {
+    html, LitElement, TemplateResult, css,
+} from 'lit';
+import { property, query } from 'lit/decorators.js';
+import type { DependentMap } from '@justeattakeaway/pie-webc-core';
+import { getSvgProps, ${sizeType} } from '@justeattakeaway/pie-icons-configs';
 
-    set size (value) {
-        this.setAttribute('size', value);
-    }
+interface IconProps {
+    size: ${sizeType};
+    class: string;
+}
 
-    get class () {
-        return this.getAttribute('class');
-    }
+const componentSelector = '${kebabCase(name)}';
 
-    set class (value) {
-        this.setAttribute('class', value);
-    }
+export class ${name} extends LitElement implements IconProps {
+    // The following styles make sure that the icon will be sized correctly
+    static styles = css\`
+        :host-context(pie-icon-button) svg,
```

Comment:
I saw @ashleynolan mentioned on his PR that :host-context isn't supported on safari or firefox so guessing this needs to be changed? Could be one for a separate PR though

Reply:
Yep good spot! I'll see what Ash comes up with then this can be updated

Reply:
I've raised a separate PR for this already, so don't worry about it as part of this PR 👍